google/cloud/pubsublite/integration_tests/publisher_integration_test.cc

```diff
@@ -0,0 +1,173 @@
+                        .value_or("")},
+        location_id_{
+            google::cloud::internal::GetEnv("GOOGLE_CLOUD_CPP_TEST_ZONE")
+                .value_or("")},
+        admin_connection_{MakeAdminServiceConnection(
+            google::cloud::internal::PopulateCommonOptions(
+                google::cloud::internal::PopulateGrpcOptions(
+                    Options{}.set<EndpointOption>(MakeLocation(location_id_)
+                                                      ->GetCloudRegion()
+                                                      .ToString() +
+                                                  "-pubsublite.googleapis.com"),
+                    ""),
+                /*endpoint_env_var=*/{}, /*emulator_env_var=*/{},
+                "pubsublite.googleapis.com"))} {
+    auto topic_id = RandomTopicName();
+
+    GarbageCollect();
+
+    CreateTopicRequest req;
+    req.set_parent("projects/" + project_id_ + "/locations/" + location_id_);
+    req.set_topic_id(topic_id);
+    req.mutable_topic()->mutable_partition_config()->set_count(3);
+    req.mutable_topic()
+        ->mutable_partition_config()
+        ->mutable_capacity()
+        ->set_publish_mib_per_sec(kThroughputCapacityMiB);
+    req.mutable_topic()
+        ->mutable_partition_config()
+        ->mutable_capacity()
+        ->set_subscribe_mib_per_sec(kThroughputCapacityMiB);
+    req.mutable_topic()->mutable_retention_config()->set_per_partition_bytes(
+        kPartitionStorage);
+    EXPECT_STATUS_OK(admin_connection_->CreateTopic(std::move(req)));
+    auto topic = Topic{project_id_, location_id_, topic_id};
+    topic_name_ = topic.FullName();
+    publisher_ = *MakePublisherConnection(
+        topic, Options{}.set<MaxBatchMessagesOption>(kMaxNumMessagesPerBatch));
+  }
+
+  ~PublisherIntegrationTest() override {
+    DeleteTopicRequest req;
+    req.set_name(topic_name_);
+    admin_connection_->DeleteTopic(std::move(req));
+  }
+
+  void GarbageCollect() {
+    ListTopicsRequest req;
+    req.set_parent("projects/" + project_id_ + "/locations/" + location_id_);
+    auto topics = admin_connection_->ListTopics(std::move(req));
+    std::string full_topic_prefix =
+        "projects/" + project_id_ + "/locations/" + location_id_ + "/topics/" +
+        TestTopicPrefix(std::chrono::system_clock::now() -
+                        std::chrono::hours(48));
+    for (auto const& topic : topics) {
+      if (!std::regex_search(topic->name(), topic_regex_)) continue;
+      if (topic->name() < full_topic_prefix) {
+        DeleteTopicRequest delete_req;
+        delete_req.set_name(topic->name());
+        admin_connection_->DeleteTopic(std::move(delete_req));
+      }
+    }
+  }
+
+  std::string RandomTopicName() {
+    auto generator =
+        google::cloud::internal::DefaultPRNG(std::random_device{}());
+    return topic_prefix_ +
+           google::cloud::internal::Sample(
+               generator, 42, "abcdefghijlkmnopqrstuvwxyz0123456789_-") +
```

Comment:
It would be nice to document where the magic `42` comes from.  Probably something to do with the length of the prefix and the maximum topic id length?